packages/nox/test/tpetra/Brusselator/Brusselator.hpp

```diff
@@ -0,0 +1,675 @@
+    // depending on the choice of Jacobian operator
+    if( overlapType == ELEMENTS ) {
+      ColumnToOverlapImporter = new timport_t(AA->getColMap(),*OverlapMap);
+      A = Teuchos::rcp(new tcrsmatrix_t(Teuchos::Copy, *AA));
+      A->FillComplete();
+    }
+
+    // Create the nodal coordinates
+    xptr = Teuchos::rcp(new tvector_t(*StandardNodeMap));
+    double Length= xmax - xmin;
+    dx=Length/((double) NumGlobalNodes-1);
+    for (i=0; i < NumMyNodes; i++) {
+      (*xptr)[i]=xmin + dx*((double) StandardNodeMap->MinMyGID()+i);
+    }
+
+    initializeSoln();
+    
+  }
+
+  // Reset problem for next parameter (time) step.
+  // For now, this simply updates oldsoln with the given Tpetra_Vector
+  void reset(const tvector_t x)
+  {
+    *oldSolution = x;
+  }
+
+  // Set initial condition for solution vector
+  void initializeSoln()
+  {
+    tvector_t& soln = *initialSolution;
+    tvector_t& x = *xptr;
+
+    // Here we do a sinusoidal perturbation of the unstable
+    // steady state.
+
+    double pi = 4.*atan(1.0);
+
+    for (int i=0; i<x.MyLength(); i++) {
+      soln[2*i] = 0.6 + 1.e-1*sin(1.0*pi*x[i]);
+      soln[2*i+1] = 10.0/3.0 + 1.e-1*sin(1.0*pi*x[i]);
+    }
+    *oldSolution = soln;
+  }
+
+  // Evaluates the function (F) and/or the Jacobian using the solution
+  // values in solnVector.
+  bool evaluate(/*NOX::Epetra::Interface::Required::FillType fType,*/ // AM: TOFIX
+                const tvector_t *soln,
+                tvector_t *tmp_rhs,
+                tvector_t *tmp_matrix)
+  {
+    /*if( fType == Tpetra::Jac ) {*/    // AM: TOFIX
+      if( overlapType == NODES ) {
+        std::cout << "This problem only works for Finite-Difference Based Jacobians"
+            << std::endl << "when overlapping nodes." << std::endl
+            << "No analytic Jacobian fill available !!" << std::endl;
+        exit(1);
+      }
+      flag = MATRIX_ONLY;
+    /*} else {*/      // AM: TOFIX
+      flag = F_ONLY;
+      if( overlapType == NODES ){
+        rhs = new tvector_t(*OverlapMap);
+      }else{
+        rhs = tmp_rhs;
+      }
+    /*}*/   // AM: TOFIX
+
+    // Create the overlapped solution and position vectors
+    tvector_t u(*OverlapMap);
+    tvector_t uold(*OverlapMap);
+    tvector_t xvec(*OverlapNodeMap);
+
+    // Export Solution to Overlap vector
+    // If the vector to be used in the fill is already in the Overlap form,
+    // we simply need to map on-processor from column-space indices to
+    // OverlapMap indices. Note that the old solution is simply fixed data that
+    // needs to be sent to an OverlapMap (ghosted) vector.  The conditional
+    // treatment for the current soution vector arises from use of
+    // FD coloring in parallel.
+    uold.doImport(*oldSolution, Importer, Tpetra::INSERT);
+    xvec.doImport(*xptr, *nodeImporter, Tpetra::INSERT);
+    if( (overlapType == ELEMENTS) /*&& (fType == Tpetra::FD_Res)*/)               // AM: TOFIX
```

Comment:
Do we have an equivalent in Tpetra for `NOX::Epetra::Interface::Required::FD_Res`?